internal/provider/resource_docker_container_funcs.go

```diff
@@ -508,6 +509,39 @@ func resourceDockerContainerCreate(ctx context.Context, d *schema.ResourceData,
 		}
 	}
 
+	if d.Get("wait").(bool) {
```

Comment:
Shouldn't this if block be inside the `if d.Get("start").(bool) {` block?
If `start: false` then the container is not spun up and then we do not need to wait whether it gets healthy.

Could you add some tests for the different possibilities?

Reply:
Right, I moved the block.